packages/cadl-ranch-specs/http/azure/core/mockapi.ts

```diff
@@ -37,24 +35,54 @@ Scenarios.Azure_Core_get = passOnSuccess(
       throw new ValidationError("Expected path param id=1", "1", req.params.id);
     }
     req.expect.containsQueryParam("api-version", "2022-12-01-preview");
-    const responseBody = { id: 1, name: "Madge" };
-    return { status: 200, body: json(responseBody) };
+    return { status: 200, body: json(validUser) };
   }),
 );
 
 Scenarios.Azure_Core_list = passOnSuccess(
   mockapi.get("/azure/core/users", (req) => {
     req.expect.containsQueryParam("api-version", "2022-12-01-preview");
+    req.expect.containsQueryParam("top", "5");
+    req.expect.containsQueryParam("skip", "10");
+    req.expect.containsQueryParam("orderby", "id");
+    req.expect.containsQueryParam("filter", "id eq 1");
+    if (!req.originalRequest.originalUrl.includes("select=id&select=orders&select=ETag")) {
+      throw new ValidationError(
+        "Expected query param colors=blue&colors=red&colors=green ",
+        "1",
+        req.headers["select"],
+      );
+    }
+    req.expect.containsQueryParam("expand", "orders");
     const responseBody = {
       value: [
-        { id: 1, name: "Madge" },
-        { id: 2, name: "John" },
+        {
+          id: 1,
+          name: "Madge",
+          ETag: "11bdc430-65e8-45ad-81d9-8ffa60d55b59",
+          orders: [{ id: 1, userId: 1, detail: "a recorder" }],
+        },
+        {
+          id: 2,
+          name: "John",
+          ETag: "11bdc430-65e8-45ad-81d9-8ffa60d55b5a",
+          orders: [{ id: 2, userId: 2, detail: "a TV" }],
+        },
       ],
```

Comment:
nit, if follow the `filter`, it should only return first item.

Reply:
changed filter sentence to be "id lt 10"